luminance/src/framebuffer.rs

```diff
@@ -68,6 +68,21 @@ use crate::backend::framebuffer::{Framebuffer as FramebufferBackend, Framebuffer
 use crate::context::GraphicsContext;
 use crate::texture::{Dim2, Dimensionable, Sampler, TextureError};
 
+struct ReprDropper<B, D>(B::FramebufferRepr)
```

Comment:
Damn. I love that idea. The concept of wrapping the repr type in a `Drop` type is really good. Thanks a lot. However, I don’t like the name of the `struct` and we need some documentation. May I suggest:

```suggestion
/// Wrapper to allow moving a framebuffer repr out of [`Framebuffer`].
struct WrappedFramebuffer<B, D>(B::FramebufferRepr)
```

Reply:
I thought it didn't need documentation since it's private to the module, but I too wasn't pleased with the name. However, `WrappedFramebuffer` doesn't really say what the wrapper is for. In a sense, the whole `pub struct Framebuffer` is also a wrapped framebuffer, no? It's wrappers and framebuffers all the way down.

My opinion (or my hope) is that it's sufficiently clear from context what this is and what it's for, since it's not visible outside this file. But you have the final say.

Reply:
One thought I had, when working on this, was that instead of having the backend expose a `destroy_framebuffer` function, couldn't the `FramebufferRepr` itself be `Drop`? To let the backend take responsibility of cleaning up after its own types, for those types that need it. That does require strict ownership enforcement though, I don't know if it would work in the general case.

Reply:
Having a method is more flexible as we might have to call it from non-Drop stuff. For framebuffers it’s not often the case but it happens for instance for shader objects.

Reply:
What is it about the name `ReprDropper` you don't like?

Reply:
The fact it doesn’t have _framebuffer_ in it. :)

Reply:
Fair enough.